### PR TITLE
Improve roi/folder action performance

### PR DESCRIPTION
**Pull request by dominikl**
Improves the performance for various ROI/Folder actions.
See [Trello - ROI Folder performance testing](https://trello.com/c/40dqTDE8/59-roi-folder-performance-testing)

**Test** Check performance with an image with 1000+ ROIs and within a group which additionally has thousands of unrelated ROIs. Check integration tests (http://regions-ci.docker.openmicroscopy.org:8080/job/OMERO-start/lastCompletedBuild/testReport/)

### CODE REVIEW

File: components/blitz/src/omero/gateway/facility/ROIFacility.java [hunk 1]
```diff
@@ -490,29 +490,24 @@ public ROIResult loadROI(SecurityContext ctx, long roiId)
...
-            
-            return result;
-        } catch (Exception e) {
-            handleException(this, e, "Cannot add ROIs to Folder ");
-            return Collections.EMPTY_MAP;
-        }
 
+            if (!toDelete.isEmpty()) {
+                Request req = Requests.delete().target("FolderRoiLink")
+                        .id(toDelete).build();
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/facility/ROIFacility.java]: Do you need a child option to prevent deleting the ROIs too?

Comment (dominikl) [inline components/blitz/src/omero/gateway/facility/ROIFacility.java]: I was wondering about that, too. Just tried without childoption and ROIs were still there :)

**Comment by dominikl** [comment]
Oh, just noticed this PR broke a couple of integration tests... Will have to adjust the tests, to not access unloaded collections.

**Comment by pwalczysko** [comment]
1000-ROIs strong fake image (plates were not in at the moment of the testing): There are max 15 seconds waiting times to add some 290 ROIs to a folder. This seems like 4x improvement at least.

**Comment by pwalczysko** [comment]
The testing of the plate with ROIs wil have to wait till tomorrow, as it takes long to import.

**Comment by pwalczysko** [comment]
Test of the plate with 100 ROIs per well, 12 folders containing all of the plate's ROIs created by the ``folderize-object.py`` script by @mtbc .
  - previously, the delay was 11 sec on opening of the ROI Tool
  - with this PR, there is no delay at all on opening of the ROI Tool
  - no other delays were observed during other interactions within the tool as a result of presence of the folders

Great improvements of performance in both cases.
Ready to merge.

**Comment by sbesson** [comment]
Great progress, thanks all.

**Pull request merged**